stable_plugins/classical_ml/data_preparation/pandas_preprocessing/routes.py

```diff
@@ -0,0 +1,361 @@
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        schema = SecondInputParametersSchema()
+
+        data_dict = dict(data)
+        fields = schema.fields
+        # define default values
+        default_values = {
+            fields["threshold"].data_key: 0,
+        }
+
+        # overwrite default values with other values if possible
+        default_values.update(data_dict)
+        data_dict = default_values
+
+        return Response(
+            render_template(
+                "second_template.html",
+                name=PDPreprocessing.instance.name,
+                version=PDPreprocessing.instance.version,
+                schema=schema,
+                values=data_dict,
+                errors=errors,
+                process=url_for(
+                    f"{PDPreprocessing_BLP.name}.FinalProcessView",
+                    db_id=db_id,
+                    step_id=step_id,
+                ),
+                add_step=url_for(
+                    f"{PDPreprocessing_BLP.name}.SecondProcessView",
+                    db_id=db_id,
+                    step_id=step_id,
+                ),
+                pandas_html=Markup(db_task.data["pandas_html"]),
+                columns_and_rows_html=Markup(db_task.data["columns_and_rows_html"]),
+                columns_as_optionlist=Markup(db_task.data["columns_as_optionlist"]),
+            )
+        )
+
+
+@PDPreprocessing_BLP.route("/<int:db_id>/<int:step_id>-process1/")
+class FinalProcessView(MethodView):
+    """Start a long running processing task."""
+
+    @PDPreprocessing_BLP.arguments(
+        SecondInputParametersSchema(unknown=EXCLUDE), location="form"
+    )
+    @PDPreprocessing_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @PDPreprocessing_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments, db_id: int, step_id: int):
+        """Start the calculation task."""
+        db_task: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        db_task.parameters = SecondInputParametersSchema().dumps(arguments)
+
+        db_task.clear_previous_step()
+        db_task.save(commit=True)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = second_task.s(
+            db_id=db_task.id, step_id=step_id
+        ) | save_task_result.s(db_id=db_task.id)
+        # save errors to db
+        task.link_error(save_task_error.s(db_id=db_task.id))
+        task.apply_async()
+
+        return redirect(
+            url_for("tasks-api.TaskView", task_id=str(db_task.id)), HTTPStatus.SEE_OTHER
+        )
+
+
+@PDPreprocessing_BLP.route("/<int:db_id>/<int:step_id>-process0/")
```

Comment:
why is this ending with 0 and not `-process-second`
also inconsistent with the ui endpoint that separates the `step_id` with a `/`